self_test/SelfTest.ts

```diff
@@ -0,0 +1,120 @@
+
+import { Injector } from 'reduct'
+import Config from '../src/services/Config'
+import { create as createLogger } from '../src/common/log'
+const ilpFetch = require('ilp-fetch')
+const log = createLogger('SelfTest')
+const WebSocket = require('ws')
+const manifestJson = require('./self-test-manifest.json')
+const axios = require('axios')
+const crypto = require('crypto')
+export default class SelfTest {
+  private config: Config
+  constructor (deps: Injector) {
+    this.config = deps(Config)
+  }
+
+  start () {
```

Comment:
It would be good if we could await the self test before declaring the codiusd instance up and running. It could be confusing if it prints that it's ready and then dies 10 seconds later because the self-test failed